### DISCUSSION TITLE
Ability to not optimise video

### DISCUSSION THREAD

**Discussion by RealOrangeOne** [photoview/photoview#131] [Ideas]
When doing a scan, if there are loads of videos, then it can take a while and be very intensive.

Personally i'm not too concerned with video, and don't mind streaming a "less than optimal" version if I need to preview it.

It'd be nice to be able to view videos, but without having to transcode them to a web format.

I'm currently getting around this by uninstalling `ffmpeg` on startup, but that means videos don't show up, which isn't ideal.

**Comment by iamjackg** [comment]
Seconding this! I just installed PhotoView today and it's fantastic, but I had to stop it because of the high ffmpeg CPU usage and huge size of the resulting cache :(

I'll try to restart it with a CPU limit in docker, but that still doesn't help with the size of the cache.

**Comment by viktorstrate** [comment]
It could also be nice to have the option to only process video formats that aren't supported in the browser.
As I think the most common video files are web compatiable, it would reduce the size of the cache significantly.
This is already how web compatiable images are handled.

This could also remove the need for `ffmpeg` being installed for web compatiable video formats. Which could be desired for small / embedded devices such as Raspberry Pis

**Comment by maltokyo** [comment]
If you have a GPU, giving ffmpeg GPU hardware acceleration would enable it to process videos in real-time (on-demand, when someone wants to watch it - otherwise it remains untouched.  This is how Jellyfin and other major home theatre apps do it).

I opened an "issue" (feature request) here: https://github.com/photoview/photoview/issues/159

**Reply by viktorstrate** [reply]
Though I think hardware acceleration wouldn't be hard to add, I don't know how hard it would be to implement on-the-fly video transcoding. 

I think it's a great idea though, and if somone has experience with this I would like to see a PoC in Go, preferably using ffmpeg, as it's already used for video transcoding.

I found this JS project that does somthing similar, though I haven't looked properly at the source code.
https://github.com/mifi/hls-vod

**Comment by maltokyo** [comment]
All I can add here is that the user does need to do some pre-setup first, depending on their video card, as here:
https://docs.linuxserver.io/images/docker-jellyfin#hardware-acceleration

After that, in Jellyfin, the container is told what the card is and given permission to use it, and it just uses it (only when it needs to, not preemptively) when there is a video format that needs transcoding to play on a web browser.
